haunter.go

```diff
@@ -0,0 +1,101 @@
+package fscache
+
+import (
+	"math"
+	"time"
+)
+
+type CacheAccessor interface {
+	FileSystemStater
+	EnumerateFiles(enumerator func(key string, f FileStream) bool)
+	RemoveFile(key string)
+}
+
+type Haunter interface {
+	Haunt(c CacheAccessor)
+	Next() time.Duration
+}
+
+type reaperHaunter struct {
+	reaper Reaper
+}
+
+type janitorHaunter struct {
+	janitor Janitor
+}
+
+type compoundHaunter struct {
+	haunters []Haunter
+}
+
+// NewCompoundHaunter returns a compound scheduleHaunt which provides a multi strategy implementation
+func NewCompoundHaunter(haunters []Haunter) Haunter {
```

Comment:
This is neat, but:
- no tests
- do we have a concrete use case?

Reply:
Decided to abandon `CompoundHaunter`.
When writing tests, it turned out that it is impossible to provide the desired result without `Scheduler` modifications. It should be taken out of the current PR.